api/src/main/java/edu/wpi/first/shuffleboard/api/sources/recording/Serialization.java

```diff
@@ -110,13 +142,28 @@ private Serialization() {
   public static void saveRecording(Recording recording, Path file) throws IOException {
     Serializers.getAdapters().forEach(a -> a.setCurrentFile(file.toFile()));
     // Work on a copy, since the recording can have new data added to it while we're in the middle of saving
-    final List<TimestampedData> dataCopy = new ArrayList<>(recording.getData());
-    recording.getData().clear();
+    var snapshot = recording.takeSnapshotAndClear();
+    final var dataCopy = new ArrayList<>(snapshot.getData());
```

Comment:
Why copy it again?

Reply:
Because it gets sorted, which `ImmutbaleList` doesn't support

Reply:
I suppose an alternative would be
```java
var dataCopy = snapshot.getData().stream().sorted().collect(ImmutableList.toImmutableList());
```